registration-checker-compiler-plugin/src/main/scala/org/virtuslab/ash/RegistrationCheckerCompilerPlugin.scala

```diff
@@ -0,0 +1,59 @@
+package org.virtuslab.ash
+
+import org.virtuslab.ash.RegistrationCheckerCompilerPlugin.cacheFileName
+
+import java.io.{File, IOException}
+import java.nio.file.{Files, NoSuchFileException}
+import java.util.function.Consumer
+import scala.collection.mutable
+import scala.tools.nsc.Global
+import scala.tools.nsc.plugins.{Plugin, PluginComponent}
+
+class RegistrationCheckerCompilerPlugin(override val global: Global) extends Plugin {
+  override val name: String = "registration-checker-plugin"
+  override val description: String =
+    "checks whether classes marked with serializability trait are being referenced in a marked serializer"
```

Comment:
Could you explain why there are two phases here? Is it just matter of code style, or is there a technical requirement for that?

Reply:
Using two phases guarantees correct ordering

Reply:
Hmmm okay... but wouldn't the same be achieved if searching for class AND THEN checking serializer were placed one after another in the same compilation phase?

Reply:
That would require either ordering source files or knowing that the current source file is the last one. We lack the necessary API  to make it work.